diff --git a/dist/runtime/handlers/renderer.mjs b/dist/runtime/handlers/renderer.mjs
index 2afdafb08f26ca1794313c846c2e738b61550679..bf9b1b085383effd2b45dca45086c04c018c7321 100644
--- a/dist/runtime/handlers/renderer.mjs
+++ b/dist/runtime/handlers/renderer.mjs
@@ -150,7 +150,9 @@ async function renderRoute(event, ssrError) {
 	}
 	
 	if (isRenderingPayload) {
-		const response = renderPayloadResponse(ssrContext);
+		const response = renderPayloadResponse(ssrContext, {
+			browserRevalidate: !import.meta.prerender && !!(routeOptions.isr || routeOptions.cache)
+		});
 		if (payloadCache) {
 			await payloadCache.setItem(ssrContext.url + ".json", response);
 		}
diff --git a/dist/runtime/utils/renderer/payload.mjs b/dist/runtime/utils/renderer/payload.mjs
index 7e10b9eb2e6bfef7234dc12eedf8c1fd6b25d22c..a07152c58d0421d9ddc3ecfe3ef145c0254b6381 100644
--- a/dist/runtime/utils/renderer/payload.mjs
+++ b/dist/runtime/utils/renderer/payload.mjs
@@ -5,15 +5,19 @@ import { stringify, uneval } from "devalue";
 import { appId, multiApp } from "#internal/nuxt.config.mjs";
 // @ts-expect-error virtual file
 import { NUXT_JSON_PAYLOADS, NUXT_NO_SSR } from "#internal/nuxt/nitro-config.mjs";
-export function renderPayloadResponse(ssrContext) {
+export function renderPayloadResponse(ssrContext, opts) {
+	const headers = {
+		"content-type": NUXT_JSON_PAYLOADS ? "application/json;charset=utf-8" : "text/javascript;charset=utf-8",
+		"x-powered-by": "Nuxt"
+	};
+	if (opts?.browserRevalidate) {
+		headers["cache-control"] = "public, max-age=0, must-revalidate";
+	}
 	return {
 		body: NUXT_JSON_PAYLOADS ? encodeForwardSlashes(stringify(splitPayload(ssrContext).payload, ssrContext["~payloadReducers"])) : `export default ${devalue(splitPayload(ssrContext).payload)}`,
 		statusCode: getResponseStatus(ssrContext.event),
 		statusMessage: getResponseStatusText(ssrContext.event),
-		headers: {
-			"content-type": NUXT_JSON_PAYLOADS ? "application/json;charset=utf-8" : "text/javascript;charset=utf-8",
-			"x-powered-by": "Nuxt"
-		}
+		headers
 	};
 }
 export function renderPayloadJsonScript(opts) {
